app/lib/actions/katello/host/destroy_host_record.rb

```diff
@@ -0,0 +1,25 @@
+module Actions
+  module Katello
+    module Host
+      class DestroyHostRecord < Actions::EntryAction
+        middleware.use ::Actions::Middleware::RemoteAction
+
+        def plan(host)
+          action_subject(host)
+          plan_self(:host_id => host.id, :hostname => host.name)
+
+          host.destroy
```

Comment:
@jlsherrill I left this in `plan` since that's where it was originally, but I could see it moving to `run`. Do you have any preference?

Reply:
The reason it was in plan was so that if there was any error it would rollback the transaction and not actually delete anything anywhere.  I've got to figure out if that still makes sense here.

Reply:
and actually, if that is all this is doing, does it even need to be an action?

Reply:
it does, without it then other tasks related to the same host that are in the task queue get really confused. for example:

1. host is created, pkg profile update task is put in
2. host is deleted without wrapping in task. At same time, pkg profile update occurs, we try to write a bunch of stuff to `katello_host_installed_packages` that references host id

In this sequence, you'll get either a "you can't write records for that host id to `host_installed_packages`, that host ID doesn't exist! 👻 " FK error, or a "you can't delete me from `hosts`, I'm still referenced in `host_installed_packages`!" constraint error.

When I put the host destroy in as a task, dynflow coordinates this.